File: src/beginner_tutorials/src/scripts/show_position.py
#!/usr/bin/env python
import rospy
from turtlesim.msg import Pose

def turtle_pose_callback(message):
    x = message.x
    y = message.y
    rospy.loginfo("Turtle Position - x: %.2f, y: %.2f"%(x,y))

def pose_listener():
    rospy.init_node("pose_listener")
    rospy.Subscriber("/turtle1/pose", Pose, turtle_pose_callback)
    rospy.spin()

if __name__ == "__main__":
    pose_listener()

File: src/beginner_tutorials/src/show_position.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"

void turtle_pose_callback(const turtlesim::Pose::ConstPtr& msg);

int main(int argc, char **argv){
    ros::init(argc, argv, "pose_listener");
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("/turtle1/pose", 1000, turtle_pose_callback);
    ros::spin();
}

void turtle_pose_callback(const turtlesim::Pose::ConstPtr& msg){
    ROS_INFO("Turtle Position - X: %.2f, Y: %.2f",msg->x, msg->y);   
    if ( (msg->x < 1.0) && (msg->y < 1.0)) {
        ROS_WARN("Turtle is near 0,0");
    }
}
